### PR TITLE
[Influxdb] handle dir changes

### PR DESCRIPTION
**Pull request by greedybro**

### CODE REVIEW

File: manala.influxdb/defaults/main.yml [hunk 1]
```diff
@@ -5,6 +5,14 @@ manala_influxdb_install_packages: ~
 manala_influxdb_install_packages_default:
   - influxdb
 
+# Dir
+manala_influxdb_meta_dir: ~
+manala_influxdb_data_dir: ~
+manala_influxdb_wal_dir: ~
+manala_influxdb_data_dir_user: influxdb
```
Comment (hyvs) [inline manala.influxdb/defaults/main.yml]: I suggest to rename it to `manala_influxdb_dir_user `, this user is used for data, meta and wal directories creation.

File: manala.influxdb/defaults/main.yml [hunk 2]
```diff
@@ -5,6 +5,14 @@ manala_influxdb_install_packages: ~
...
+manala_influxdb_data_dir_group: influxdb
```
Comment (hyvs) [inline manala.influxdb/defaults/main.yml]: Same here.

File: manala.influxdb/defaults/main.yml [hunk 3]
```diff
@@ -5,6 +5,14 @@ manala_influxdb_install_packages: ~
 manala_influxdb_install_packages_default:
   - influxdb
 
+# Dir
+manala_influxdb_meta_dir: ~
```
Comment (hyvs) [inline manala.influxdb/defaults/main.yml]: What do you thing to regroup all dir variables under a `manala_influxdb_directories` list ? 

We should use only one file task : 
```
- name: dir > Create directory
  file:
    path: "{{ item }}"
    state: directory
    owner: "{{ manala_influxdb_directories_user }}"
    group: "{{ manala_influxdb_directories_group }}"
    mode: "{{ manala_influxdb_directories_mode }}"
  with_items: "{{ manala_influxdb_directories }}"
```

Comment (greedybro) [reply manala.influxdb/defaults/main.yml]: Done. It is cleaner indeed !

Comment (greedybro) [reply manala.influxdb/defaults/main.yml]: although we can't do that with only one variable :

```
    manala_influxdb_config:
      - reporting-disabled: true
      - meta:
        - dir: "{{ manala_influxdb_meta_dir }}"
      - data:
        - dir: "{{ manala_influxdb_data_dir }}"
        - wal-dir: "{{ manala_influxdb_wal_dir }}" 
```

Comment (hyvs) [reply manala.influxdb/defaults/main.yml]: IMHO you should not reference directly role variables in your playbook ; instead you should do something like that : 

```
_influxdb_dirs:
 data: /mnt/influxdb/data
 wal: /mnt/influxdb/wal

manala_influxdb_directories: "{{ _influxdb_dirs.values() | list }}"
manala_influxdb_config:
  - meta: 
    - dir: "{{ _influxdb_dirs.dir }}"
  - data:
    - dir: "{{ _influxdb_dirs.data }}"
    - wal-dir: "{{ _influxdb_dirs.wal_dir }}"
```

File: manala.influxdb/tasks/dir.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: dir > Create directory
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ manala_influxdb_dir_user }}"
+    group: "{{ manala_influxdb_dir_group }}"
+    mode: "{{ manala_influxdb_dir_mode }}"
+  with_items: "{{ manala_influxdb_dir}}"
```
Comment (hyvs) [inline manala.influxdb/tasks/dir.yml]:
```suggestion
  with_items: "{{ manala_influxdb_dir }}"
```

File: manala.influxdb/tasks/dir.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  when: manala_influxdb_dir|default(false)
```
Comment (hyvs) [inline manala.influxdb/tasks/dir.yml]: Remove this condition, its already handled by empty list on with_items lookup

File: manala.influxdb/templates/config/base.conf.j2 [hunk 6]
```diff
@@ -197,12 +197,12 @@
   # Values without a size suffix are in bytes.
   {{ macros.config_row(config_data, 'max-index-log-file-size', '# max-index-log-file-size = "1m"', 0, true) }}
 
-  # The size of the internal cache used in the TSI index to store previously
```
Comment (hyvs) [inline manala.influxdb/templates/config/base.conf.j2]: Dont alter config template, its iso with upstream template, we need to keep extra spaces.

File: manala.influxdb/tests/0400_services.yml [hunk 7]
```diff
@@ -7,7 +7,7 @@
     - import_tasks: pre_tasks/disable_systemd.yml
     - import_tasks: pre_tasks/influxdata.yml
   roles:
-    - manala.influxdb
+    - role: manala.influxdb
```
Comment (hyvs) [inline manala.influxdb/tests/0400_services.yml]: keep previous syntax

**Pull request merged**